File: Linked List/2130. Maximum Twin Sum of a Linked List.py
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        n = 0
        cp_head = head
        while cp_head:
            n += 1
            cp_head = cp_head.next

        i = 0
        store = defaultdict(int)
        while head:
            key = i
            if i < (n // 2):
                key = n - i - 1
            store[key] += head.val
            i += 1
            head = head.next

        return max(store.values())

File: Linked List/2130. Maximum Twin Sum of a Linked List.cpp
class Solution {
public:
    int pairSum(ListNode* head) 
    {
        vector<long long>v;
        while(head)
        {
            v.push_back(head->val);
            head = head->next;
        }
        long long ans = 0;
        int i = 0, j = v.size()-1;
        while(i<j)
        {
            ans = max(ans, v[i]+v[j]);
            i++, j--;
        }
        return ans;
    }
};